src/osp/Resource/resources.cpp

```diff
@@ -71,6 +71,13 @@ ResId Resources::find(ResTypeId typeId, PkgId pkgId, std::string_view name) cons
     return findIt->second;
 }
 
+std::string_view Resources::name(ResTypeId typeId, ResId resId) const noexcept
+{
+    PerResType const &rPerResType = get_type(typeId);
+
+    return rPerResType.m_resNames[std::size_t(resId)];
```

Comment:
```


Check warning on line 78 in src/osp/Resource/resources.cpp

Code scanning

Do not slice (es.63).

Warning
Do not slice (es.63).
Show more details 
```

This function should return `decltype(rPerResType.m_resNames[std::size_t(resId)])`

Reply:
So this is osp::SharedString being casted to osp::string_view; apparently that brings up this warning?

Reply:
Imho you should preserve the type so that consumers of the funciton's return'ed value can choose wether they want to treat it as a string view, or treat it as a shared string